dds/DCPS/transport/rtps_udp/RtpsUdpDataLink.cpp

```diff
@@ -4323,7 +4331,10 @@ RtpsUdpDataLink::RtpsReader::deliver_held_data(const RepoId& src)
     const SequenceNumber ca = wi->second->recvd_.cumulative_ack();
     const WriterInfo::HeldMap::iterator end = wi->second->held_.upper_bound(ca);
     for (WriterInfo::HeldMap::iterator it = wi->second->held_.begin(); it != end; /*increment in loop body*/) {
-      to_deliver.push_back(it->second);
+      const DCPS::SystemTimePoint source_time(it->second.header_.get_source_timestamp());
```

Comment:
Am I correct in that you are going filter on both sides?

Reply:
Yes, current plan is writer side filters based on association with remote readers, and reader side filters based on reader creation time (which should hopefully only come into play for interop).